zone/parcels.cpp

```diff
@@ -0,0 +1,755 @@
+/*	EQEMu: Everquest Server Emulator
+	Copyright (C) 2001-2002 EQEMu Development Team (http://eqemu.org)
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; version 2 of the License.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY except by those people which sell it, which
+	are required to give you total support for your newly bought product;
+	without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+	A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with this program; if not, write to the Free Software
+	Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+*/
+
+#include "../common/global_define.h"
+#include "../common/events/player_event_logs.h"
+#include "../common/repositories/trader_repository.h"
+#include "worldserver.h"
+#include "string_ids.h"
+#include "parcels.h"
+#include "client.h"
+#include "../common/ruletypes.h"
+
+extern WorldServer worldserver;
+
+void Client::SendBulkParcels()
+{
+	SetEngagedWithParcelMerchant(true);
+	LoadParcels();
+
+	if (m_parcels.empty()) {
+		return;
+	}
+
+	ParcelMessaging_Struct pms{};
+	pms.packet_type = ItemPacketParcel;
+
+	std::stringstream           ss;
+	cereal::BinaryOutputArchive ar(ss);
+
+	for (auto &p: m_parcels) {
+		auto item = database.GetItem(p.second.item_id);
+		if (item) {
+			auto inst = database.CreateItem(item, p.second.quantity);
```

Comment:
`inst` needs to be deleted

Reply:
As noted above, converted to unique_ptr.